### PR TITLE
chore: Refactor repo limit check function

### PR DESCRIPTION
**Pull request by AnaghHegde**
## How Has This Been Tested?
- Locally

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java [hunk 1]
```diff
@@ -399,18 +399,12 @@ public Mono<String> commitApplication(GitCommitDTO commitDTO, String defaultAppl
...
                                     return Mono.just(defaultApplication);
                                 }
-                                return this.getApplicationCountWithPrivateRepo(workspaceId)
-                                        .map(privateRepoCount -> {
-                                            if (limit >= privateRepoCount) {
-                                                return defaultApplication;
-                                            }
-                                            throw new AppsmithException(AppsmithError.GIT_APPLICATION_LIMIT_ERROR);
-                                        });
+                                throw new AppsmithException(AppsmithError.GIT_APPLICATION_LIMIT_ERROR);
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:407]: Minor Q : Any reason why we are not sending analytics events in this scenario but seem to be doing that for the rest of the instances where repo limit has been reached?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java [hunk 2]
```diff
@@ -2457,4 +2434,20 @@ FieldName.ORGANIZATION_ID, defaultIfNull(application.getWorkspaceId(), ""),
                     return application;
                 });
     }
+
+    private Mono<Boolean> isRepoLimitReached(String workspaceId, Boolean isClearCache) {
+        return gitCloudServicesUtils.getPrivateRepoLimitForOrg(workspaceId, isClearCache)
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:2439]: We don't seem to be using the `isClearCache` variable. Can you help me understand the variable and how it was being used earlier? Thanks!

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java (hunk 1 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:407]: This is a commit flow. This is a hack where the user makes the repo public while connecting but changes this to private and the commit will fail. So we decided this is not required in analytics.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java (hunk 2 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:2439]: This is used in the util class where we decide if we need to fetch the limit from the cloud service or use it from the cache. For connect and import flow we fetch the data from CS but for commit(scenario mentioned in the above comment) we use the value from the cache.

**Review by trishaanand — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java [hunk 3]
```diff
@@ -2457,4 +2434,20 @@ FieldName.ORGANIZATION_ID, defaultIfNull(application.getWorkspaceId(), ""),
                     return application;
                 });
     }
+
+    private Mono<Boolean> isRepoLimitReached(String workspaceId, Boolean isClearCache) {
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:2438]:
```suggestion
    protected Mono<Boolean> isRepoLimitReached(String workspaceId, Boolean isClearCache) {
```

**Review by trishaanand — APPROVED** [review]

**Review by trishaanand — APPROVED** [review]

**Pull request merged**